common/views/components/IIIFPresentationPreview/IIIFPresentationPreview.js

```diff
@@ -1,199 +1,156 @@
-  &::before {
-    left: 7px;
-    top: 7px;
-    z-index: -1;
-  }
-
-  /* Third sheet of paper */
-  &::after {
-    left: 14px;
-    top: 14px;
-    z-index: -2;
-  }
+  const numberOfImages = canvases.length < n ? canvases.length : n;
 
-  img {
-    display: block;
+  for (var i = 1; i <= numberOfImages; i++) {
+    const randomNumber = Math.floor(Math.random() * canvases.length);
+    const randomCanvas = canvases.splice(randomNumber, 1)[0];
+    images.push({
+      orientation:
+        randomCanvas.thumbnail.service.width >
+        randomCanvas.thumbnail.service.height
+          ? 'landscape'
+          : 'portrait',
+      uri: iiifImageTemplate(randomCanvas.thumbnail.service['@id'])({
+        size: '!400,400',
+      }),
+      canvasId: randomCanvas['@id'],
+    });
   }
-`;
+  return {
+    label: 'random',
+    images,
+  };
+}
 
-const ScrollContainer = styled.div`
-  overflow-x: scroll;
+function structuredImages(iiifManifest = null) {
+  const structures = iiifManifest ? iiifManifest.structures : [];
+  return structures.map(structure => {
+    const images = structure.canvases.map(canvasId => {
+      const matchingCanvas =
+        iiifManifest &&
+        iiifManifest.sequences
+          .find(sequence => sequence['@type'] === 'sc:Sequence')
+          .canvases.find(canvas => {
+            return canvas['@id'] === canvasId;
+          });
+      return (
+        matchingCanvas && {
+          orientation:
+            matchingCanvas.thumbnail.service.width >
+            matchingCanvas.thumbnail.service.height
+              ? 'landscape'
+              : 'portrait',
+          uri: iiifImageTemplate(matchingCanvas.thumbnail.service['@id'])({
+            size: '!400,400',
+          }),
+          canvasId,
+        }
+      );
+    });
+    return {
+      label: structure.label,
+      images,
+    };
+  });
+}
 
-  &::-webkit-scrollbar {
-    margin-top: ${props => `${props.theme.spacingUnit}px`};
-    height: ${props => `${props.theme.spacingUnit * 5}px`};
-    background: ${props => props.theme.colors.cream};
-  }
-
-  &::-webkit-scrollbar-thumb {
-    border-radius: 0;
-    border-style: solid;
-    border-width: ${props => `${props.theme.spacingUnit * 2}px 0`};
-    border-color: ${props => props.theme.colors.cream};
-    background: ${props => props.theme.colors.marble};
-  }
+function orderedStructuredImages(structuredImages) {
+  const titlePage = structuredImages.find(
+    structuredImage => structuredImage.label === 'Title Page'
+  );
+  const frontCover = structuredImages.find(
+    structuredImage =>
+      structuredImage.label === 'Front Cover' ||
+      structuredImage.label === 'Cover'
+  );
+  const firstTableOfContents = structuredImages.find(
+    structuredImage => structuredImage.label === 'Table of Contents'
+  );
+  return [titlePage, frontCover, firstTableOfContents].filter(Boolean);
+}
 
-  & > div {
-    display: flex;
-  }
-`;
+function previewThumbnails(
+  iiifManifest = {},
```

Comment:
Where do we need these defaults? It can imply that there might be a useful default return, which in this case, it doesn't seem to be. We can always do null checking in the caller to keep functions tidier.